centreon/src/Core/Security/Token/Application/UseCase/AddToken/AddToken.php

```diff
@@ -0,0 +1,166 @@
+use Core\Security\Token\Application\Repository\ReadTokenRepositoryInterface;
+use Core\Security\Token\Application\Repository\WriteTokenRepositoryInterface;
+use Core\Security\Token\Domain\Model\NewToken;
+use Security\Encryption;
+
+final class AddToken
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private readonly WriteTokenRepositoryInterface $writeTokenRepository,
+        private readonly ReadTokenRepositoryInterface $readTokenRepository,
+        private readonly ReadContactRepositoryInterface $readContactRepository,
+        private readonly ProviderAuthenticationFactoryInterface $providerFactory,
+        private readonly AddTokenValidation $validation,
+        private readonly ContactInterface $user,
+    ) {
+    }
+
+    /**
+     * @param AddTokenRequest $request
+     * @param AddTokenPresenterInterface $presenter
+     */
+    public function __invoke(AddTokenRequest $request, AddTokenPresenterInterface $presenter): void
+    {
+        try {
+            if (! $this->user->hasTopologyRole(Contact::ROLE_ADMINISTRATION_API_TOKENS_RW)) {
+                $this->error(
+                    "User doesn't have sufficient rights to add a token",
+                    ['user_id' => $this->user->getId()]
+                );
+                $presenter->presentResponse(
+                    new ForbiddenResponse(TokenException::addNotAllowed()->getMessage())
+                );
+
+                return;
+            }
+
+            $tokenString = $this->createToken($request);
+
+            $presenter->presentResponse(
+                $this->createResponse($tokenString)
+            );
+        } catch (AssertionFailedException|\ValueError $ex) {
+            $presenter->presentResponse(new InvalidArgumentResponse($ex));
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+        } catch (TokenException $ex) {
+            $presenter->presentResponse(
+                match ($ex->getCode()) {
+                    TokenException::CODE_CONFLICT => new ConflictResponse($ex),
+                    default => new ErrorResponse($ex),
+                }
+            );
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+        } catch (\Throwable $ex) {
+            $presenter->presentResponse(
+                new ErrorResponse(TokenException::addToken())
+            );
+            $this->error((string) $ex);
+        }
+    }
+
+    /**
+     * @param AddTokenRequest $request
+     *
+     * @throws AssertionFailedException
+     * @throws TokenException
+     * @throws \Throwable
+     *
+     * @return string
+     */
+    private function createToken(AddTokenRequest $request): string
+    {
+        $this->validation->assertIsValidName($request->name);
+        $this->validation->assertIsValidUser($request->userId);
+
+        $localProviderId = $this->providerFactory->create(Provider::LOCAL)->getConfiguration()->getId();
```

Comment:
No longer needed

Reply:
Has been removed from Token model, still used in the newToken model